src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden !== true && animal instanceof Herbivore) {
+      animal.health = animal.health - 50;
+    }
+
+    if (animal.health === 0) {
+      const indexOfDeletedAnimal = Animal.alive.indexOf(animal);
+
+      if (indexOfDeletedAnimal > -1) {
+        Animal.alive.splice(indexOfDeletedAnimal, 1);
+      }
+    }
```

Comment:
тут можно сделать проще:
```suggestion
    Animal.alive = Animal.alive.filter(callback....)
```